### PR TITLE
feat: Remove unidirectional networks - no longer supported

### LINKED ISSUE #11796: Remove unidirectional networks - no longer supported.
As part of #11506, get rid of unidirectional network code as it's no longer supported.

### PR DESCRIPTION
**Pull request by kfa-aguda**
Closes #11796

### CODE REVIEW

**Review by anthony-swirldslabs — DISMISSED** [review]
Platform Data changes look good.

**Review by OlegMazurov — APPROVED** [review]

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java [hunk 1]
```diff
@@ -129,8 +129,9 @@ void testFullyConnectedUnidirectionalTopology(final int numNodes, final int numN
                     .toList();
             assertEquals(expected, neighbors, "all should be neighbors except me");
             for (final NodeId neighbor : neighbors) {
-                assertTrue(topology.shouldConnectTo(neighbor), "I should connect to all neighbors");
-                assertTrue(topology.shouldConnectToMe(neighbor), "all neighbors should connect to me");
+                assertTrue(
+                        topology.shouldConnectTo(neighbor) ^ topology.shouldConnectToMe(neighbor),
+                        "I should connect to all neighbors but not all neighbors should connect to me & vice versa");
```
[the remark this answers is not in the document]
Comment (kfa-aguda) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java:134]: Good catch. I aimed to keep the old comment & at the same time mirror the XOR relationship, but it ended up not being the best. Thanks for the suggestion.

[the remark this answers is not in the document]
Comment (kfa-aguda) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/TopologyTest.java]: Applied now.

**Comment by lpetrovic05** [comment]
@kfa-aguda can you also edit the docs to remove everything related to unidirectional networks?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/AbstractGossip.java [hunk 2]
```diff
@@ -141,8 +141,7 @@ protected AbstractGossip(
         final CryptoConfig cryptoConfig = platformContext.getConfiguration().getConfigData(CryptoConfig.class);
         final SocketConfig socketConfig = platformContext.getConfiguration().getConfigData(SocketConfig.class);
 
-        topology = new StaticTopology(
-                addressBook, selfId, basicConfig.numConnections(), unidirectionalConnectionsEnabled());
+        topology = new StaticTopology(addressBook, selfId, basicConfig.numConnections());
```
Comment (mxtartaglia-sl) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/AbstractGossip.java:144]: I don't see where the topology is changed later in the code; if I didn't miss it, maybe declare as  final ?

Comment (kfa-aguda) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/AbstractGossip.java:144]: topology is already final. This is old code, I just updated the params to reflect the new constructor signature.

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/StaticConnectionManagersTest.java [hunk 3]
```diff
@@ -57,47 +58,44 @@ void test(final int numNodes, final int numNeighbors) throws Exception {
         final AddressBook addressBook =
                 new RandomAddressBookGenerator(r).setSize(numNodes).build();
         final NodeId selfId = addressBook.getNodeId(r.nextInt(numNodes));
-        Mockito.when(connectionCreator.createConnection(Mockito.any())).thenAnswer(inv -> {
+        lenient().when(connectionCreator.createConnection(Mockito.any())).thenAnswer(inv -> {
```
[the remark this answers is not in the document]
Comment (kfa-aguda) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/network/StaticConnectionManagersTest.java]: fixed

**Review by mxtartaglia-sl — APPROVED** [review]

**Review by cody-littley — APPROVED** [review]

**Review by anthony-swirldslabs — APPROVED** [review]

**Pull request merged**